pkg/reconciler/common/prune.go

```diff
@@ -279,22 +280,42 @@ func (pruner *Pruner) checkAndCreate(ctx context.Context, uniquePruneNs, schedul
 				return err
 			}
 		}
-		return createCronJob(ctx, pruner.kc, suffixedCronName, pruner.targetNamespace, uniquePruneNs, schedule, pruneContainers, pruner.ownerRef)
+		return createCronJob(ctx, pruner.kc, suffixedCronName, pruner.targetNamespace, uniquePruneNs, schedule, pruneContainers, pruner.ownerRef, tC)
 	}
 
 	// no change in config but the cronjob does not exist
 	if len(cronList.Items) == 0 && !configChanged {
-		return createCronJob(ctx, pruner.kc, suffixedCronName, pruner.targetNamespace, uniquePruneNs, schedule, pruneContainers, pruner.ownerRef)
+		return createCronJob(ctx, pruner.kc, suffixedCronName, pruner.targetNamespace, uniquePruneNs, schedule, pruneContainers, pruner.ownerRef, tC)
 	}
 
 	// any case with config change
 	if configChanged {
-		return createCronJob(ctx, pruner.kc, suffixedCronName, pruner.targetNamespace, uniquePruneNs, schedule, pruneContainers, pruner.ownerRef)
+		return createCronJob(ctx, pruner.kc, suffixedCronName, pruner.targetNamespace, uniquePruneNs, schedule, pruneContainers, pruner.ownerRef, tC)
 	}
+
 	return nil
 }
+func getPodSpec(pruneContainers []corev1.Container, tC *v1alpha1.TektonConfig) corev1.PodSpec {
+	targetNamespace := tC.Spec.TargetNamespace
+
+	if targetNamespace == "openshift-pipelines" {
```

Comment:
i see that we are using this check to determine if the platform is openshift. 
This is not a highly reliable way to check if the platform is openshift as users or olm dependency resolution can install the operator in a different namespace.

Instead, we should check for a presence of an OpenShift specific Resource for example any CRD that is only seen in OpneShift like ConsoleCLI, SampleYamls etc.

Reply:
cc @pradeepitm12 @piyush-garg

Reply:
Yes not very reliable way, let me change this.

Reply:
Why do we want to do it only on openshift? we have the feature of providing the node selector and tolerations on config in both openshift and k8s afaik,

Reply:
@rupalibehera this pr https://github.com/tektoncd/operator/pull/710 [here](https://github.com/tektoncd/operator/pull/710/files#diff-93757650a499e93f88d69c44f58be32b522e5ec5f200aa5c77aee19f75b353d0R105)
adds a way to identify a platform in a simple way. 
so once that merges, any controller will be able to make a simple fucntion call and find the current platform.

Reply:
could you hold this pr until #710 merges

Reply:
@piyush-garg, earlier I thought we need it only for openshift, it does make sense to remove this check completely let me update the PR.

Reply:
> could you hold this pr until #710 merges

yeah, but I think so as @piyush-garg mentioned we can remove this check